test/sam.c

```diff
@@ -982,11 +984,14 @@ static void samrecord_layout(void)
 
     size_t bam1_t_size, bam1_t_size2;
 
-    bam1_t_size = 36 + sizeof(int) + 4 + sizeof (char *) + sizeof(uint64_t);
+    assert(sizeof(hts_pos_t) == 8 || sizeof(hts_pos_t) == 4);
+    int core_size = sizeof(hts_pos_t) == 8 ? 48 : 36;
+    bam1_t_size = core_size + sizeof (int) + 4 + sizeof (char *) + sizeof(uint64_t);
```

Comment:
Although a test code, this looks a bit ugly. Maybe it can be changed to something like:
```c
bam1_t_size = core_size + sizeof (int) + sizeof (char *) + sizeof (uint64_t) + (sizeof (uint32_t) << 1);
```
This includes the newly added `mempolicy` member.